### PR TITLE
[DOCS] Document GridMap

### PR DESCRIPTION
**Pull request by dsaltares**

### CODE REVIEW

File: doc/classes/GridMap.xml [hunk 1]
```diff
@@ -0,0 +1,199 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="GridMap" inherits="Spatial" category="Cpre" version="3.0.alpha.custom_build">
+	<brief_description>
+		Node for 3D tile-based maps.
+	</brief_description>
+	<description>
+		Node for 3D tile-based maps.
+
```
Comment (NathanLovato) [inline doc/classes/GridMap.xml]: 1 line = 1 new empty paragraph. Gotta remove them.

File: doc/classes/GridMap.xml [hunk 2]
```diff
@@ -0,0 +1,199 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="GridMap" inherits="Spatial" category="Cpre" version="3.0.alpha.custom_build">
+	<brief_description>
+		Node for 3D tile-based maps.
+	</brief_description>
+	<description>
+		Node for 3D tile-based maps.
```
Comment (NathanLovato) [inline doc/classes/GridMap.xml]: Don't duplicate the brief description, try to develop it instead. E.g. (note, I don't know how the node works, it's just an imaginary example)
`[code]GridMap[/code] lets you place meshes on a grid interactively. It works both from the editor and helps you create in-game level editors.`

File: doc/classes/GridMap.xml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		
+		GridMaps use a [MeshLibrary] which contain a list of tiles (meshes with materials plus optional collisions and extra elements) which can be used to create grid-based maps in 3D.
+
+		A GridMap contains a collection of Cells. Each grid Cell refers to a [MeshLibrary] item. All Cells in the map have the same dimensions.
+
+		A GridMap is split into a sparse collection of Octants for efficient rendering and physics processing. Every Octant has the same dimensions and can contain several Cells.
```
Comment (NathanLovato) [inline doc/classes/GridMap.xml]: Remove caps on cells if they're just a concept. If it's some member variable, either link to it [member var_name] or use [code][/code]

**Review by NathanLovato — CHANGES_REQUESTED** [review]
@reduz @akien-mga The XML didn't exist for some reason, although the node does exist in the editor. Can someone look into this and generate the XML? Seems we're missing <member> tags.

File: doc/classes/GridMap.xml (hunk 3 above)
Comment (saltares) [reply doc/classes/GridMap.xml]: They are technically private inner classes. As far as the user is concerned they don't exist in code, so I assume the correct way is treat them as concepts (lowercase).

**Comment by saltares** [comment]
@NathanLovato I think the comments have been addressed.

**Comment by groud** [comment]
> @reduz @akien-mga The XML didn't exist for some reason, although the node does exist in the editor. Can someone look into this and generate the XML? Seems we're missing tags.

The documentation for the GridMap class is in the gripmap module folder (/modules/gridmap). It has a config.py file with two functions: get_doc_classes() and get_doc_path() that return a path relative to the module path. Thus the documentation is stored there.

Either we have:
- the doc should stay here and the config.py file should then be modified
- or the documentation should stay in the module folder and this PR is not good.

**Review by NathanLovato — APPROVED** [review]

**Comment by NathanLovato** [comment]
Approved, but not merging yet. What seems best to you? To me, it'd be nice if all the docs files could be in the same folder considering they're all part of the docs. This way the classref status tool can list absolutely everything with a simple command.

**Comment by groud** [comment]
I agree. I would just like to know if there is any reason why this doc was here. @akien-mga maybe ?

**Comment by akien-mga** [comment]
Modules are meant to be documented in `modules/<name>/doc_classes/`, and GridMap is (for now) the only one in the upstream repo using this scheme, but any other thirdparty module can do it too. Modules are meant to be self-contained, so we should not document them in `doc/classes/`.

The template exists already: https://github.com/godotengine/godot/blob/master/modules/gridmap/doc_classes/GridMap.xml

Please move your docs there, and squash commits together.

**Review by akien-mga — CHANGES_REQUESTED** [review]
Needs squash and move to `modules/gridmap/doc_classes/GridMap.xml`.

**Comment by groud** [comment]
The config.py file should be also updated (or removed maybe ?) then.

**Comment by akien-mga** [comment]
> The config.py file should be also updated (or removed maybe ?) then.

What config.py file?

**Comment by groud** [comment]
> What config.py file?

I was thinking about the one in `modules/gridmap/`. 

But in fact you are right, I didn't read your message correctly, it does not need to be modified.

**Comment by akien-mga** [comment]
Growing stale, I guess I'll do the `git mv` myself.

**Comment by akien-mga** [comment]
Fixed and merged as 62ba7a1eb.

**Pull request closed without merging**